ebmlite/core.py

```diff
@@ -809,7 +809,7 @@ def __init__(self,
                 regardless, and stored in the Document's `info` attribute.
         """
         self._ownsStream = False
-        if isinstance(stream, (str, bytes, bytearray)):
+        if isinstance(stream, (str, bytes, bytearray, Path)):
```

Comment:
Checking for `bytes` and `bytearray` filenames probably isn't necessary... could you remove these, here and elsewhere?

Also, in the `__init__()` definition, could you change the annotation from `BinaryIO` to `Union[str, Path, BinaryIO]`